proposals/operator-ingress-sketch.md

```diff
@@ -0,0 +1,189 @@
+# Operator Ingress API sketch
+
+- I want simple things to be easy, harder things should still to be possible.
+- I want to be able to get a usable proxy up and running using as little yaml as possible.   I value
+  this because I thik it give a compelling demo that people can follow.
+
+## Roles
+
+* Infrastructure admin - responsible for the proxy tier and how it exposes itself to the network.
+* Developer - responsible for declaraction of the virtualcluster
+
+## API
+
+### Proxy
+
+Owned by the Infrastructure admin.
+
+Declares an instance of the proxy.  It is valid for this to be an empty object.  The empty object
+significes a proxy with a single replica which is exposed on-cluster using a plain transport.
+
+```yaml
+apiVersion: proxy.kroxylicious.io/v1
+kind: Proxy
+metadata:
+  name: myproxy
+spec:
+   # All fields are optional - an empty proxy is sufficient to get going
+
+   infrastructure:
+      replics: 2
+      # resources / env
+      # labels / annotations to be applied.
+
+   # Refereneces one or more ingresses.
+   # If no ingresses are defined, the operator will behave as if a plain clusterip ingress is defined called default.
+   ingresses:
+   - myclusterip
+   - foobar
+status:
+   # reports ingresses
+```
+
+# ProxyIngress
+
+Owned by the Infrastructure admin.
+
+Declares how the proxy exposes itselt within the cluster or to the outside world.
+
+
+```yaml
+apiVersion: proxy.kroxylicious.io/v1
+kind: ProxyIngress
+metadata:
+  name: myclusterip
+spec:
+  # oneOf
+  service:
+    type: ClusterIP|LoadBalancer
+    brokerAddressPattern: $(node)-${vitrtualClusterName}.example.com  (LoadBalancer only)
+    protocol: TCP|TLS (ClusterIP supports both, LoadBalancer TLS only)
+    port: # optional, defaults to 9082
+  openShiftRoute:
+    ingressControllName: # ingress controller name (optional)
+  gateway:
+    type: TLSRoute
+    # think about this
+```      
+
+# VirtualCluster
+
+Created by a developer
+
+```yaml
+apiVersion: proxy.kroxylicious.io/v1
+kind: VirtualCluster
+metadata:
+  name: mycluster
+spec:
+  proxyReference:
+    kind: ProxyIngress
+    group: proxy.kroxylicious.io
+    name: myproxy
```

Comment:
Does this point to the Proxy, or the ProxyIngress (the `name` and the `type` disagree with the above yamls, so it's not clear).

Reply:
That was a typo.  The VirtualCluster refers to the Proxy.